### ISSUE TITLE
Document connection strings for unix sockets with postgres

### ISSUE THREAD

**Issue by lf-** [rust-db/refinery#258]
I am using `tmp-postgres` for Haskell which creates a Unix socket with a connection string like `host=/tmp/tmp-postgres-socket-33d7937af9c67679 dbname=postgres port=49905`.

I want to use refinery-cli to apply my migrations to such a database.

According to the postgres docs, the connection string can be in several formats, which *do* support Unix sockets: https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING

However, it seems that Refinery parses them itself:

https://github.com/rust-db/refinery/blob/36e2c219de236c95853a0aa6cd606a2d52d255f8/refinery_core/src/config.rs#L225-L243

It *seems* like I need to generate a connection string like so, but this was obtained largely by guessing:

```haskell
{- | Serializes a 'PG.Options' into the right format for Refinery CLI, which is

 > postgres://user:pass@(urlencoded host)/db

 See <https://github.com/rust-db/refinery/blob/36e2c219de236c95853a0aa6cd606a2d52d255f8/refinery_core/src/config.rs#L225-L243>
-}
optionsToUri :: PG.Options -> Maybe (URIRef Absolute)
optionsToUri PG.Options {host, user, password, dbname, port} = do
  host' <- getLastBS host
  dbname' <- getLastBS dbname
  port' <- getLast port
  -- Postgres will be grumpy if you don't give it a user, so there's not any
  -- reasonable default here
  user' <- getLastBS user
  pure $
    URI
      { uriScheme = Scheme "postgres"
      , uriAuthority =
          Just $
            Authority
              { authorityUserInfo =
                  Just $ UserInfo user' (fromMaybe "" $ getLastBS password)
              , authorityHost = Host (cs . toLazyByteString $ urlEncode [] host')
              , authorityPort = Just $ Port port'
              }
      , uriPath = "/" <> cs dbname'
      , uriQuery = Query []
      , uriFragment = Nothing
      }
  where
    getLastBS :: Last String -> Maybe ByteString
    getLastBS s = cs <$> getLast s
```

**Comment by jxs** [comment]
Hi, and thanks for the Interest! So, yeah `refinery` assumes you pass a connection uri, defined in the doc [you linked above ](https://www.postgresql.org/docs/current/libpq-connect.html#id-1.7.3.8.3.6). If I understand correctly it worked after you did so right? Would you be willing to submit a PR improving the documentation to make users aware of it?
Thanks!

**Comment by lf-** [comment]
that's the thing: it does not parse those as far as i can tell. it only takes the one format that i guessed by reading the code since it does the parsing itself rather than giving the whole thing to postgres. 

if i remember, I'll look into filing a pr on Monday

**Comment by jxs** [comment]
yeah, but it then returns a connection string to postgres, see [here](https://github.com/rust-db/refinery/blob/36e2c219de236c95853a0aa6cd606a2d52d255f8/refinery_core/src/drivers/config.rs#L82-L83) should have the same format as the input doesn't it on your case?

**Comment by lf-** [comment]
Ah. Yes, but most of the formats supported by postgres are thus unsupported. For instance refinery seemingly discards query parameters.

**Comment by jxs** [comment]
Ah, I see. I will gladly support them for the drivers that support them on the input urls if you want to submit a PR :)

**Comment by lf-** [comment]
I HATE STALE BOT. but also, yeah no this is still a bug, i wrote a very dubious socat workaround for this last week. unfortunately i have no time to fix it properly :(

**Comment by lf-** [comment]
MISCLICKED. This is still a bug.

**Comment by jxs** [comment]
> I HATE STALE BOT. but also, yeah no this is still a bug, i wrote a very dubious socat workaround for this last week. unfortunately i have no time to fix it properly :(

Hi, I feel this is more of a lack of feature and not a bug.
The issue was closed as it didn't have progress, but you very welcome to submit a PR to address this.
Thanks
